### ISSUE TITLE
Automated testing framework for IMPROVE

### ISSUE THREAD

**Issue by adpartin** [JDACS4C-IMPROVE/IMPROVE#201]
Contributors: @rajeeja (lead), @CesarAugusto14, @Gayara-Fernando 

**Development of automated testing framework**
To maintain robustness as we continue developing improvelib and the associated workflows, we need to implement an automated testing framework that ensures that modifications to improvelib or workflows do not introduce bugs. The goal is to develop a system to execute small-scale versions of workflows on curated models whenever updates are made to the workflows or improvelib.

**What is small-scale?**
The definition of “small-scale” may vary for each workflow. For example:
CSA. Datasets: CCLE and gCSI. Splits: 0, 1
LCA. Datasets: CCLE or gCSI. Splits: 0. Train sizes: 5 sizes.
HPO. Datasets: gCSI. Splits: 0. HPs: batch size, learning rate. Range: TODO

**Automated tests**
The details of how tests are run should make sense. Here is a non-exhaustive list of a few examples:
If parsl_csa workflow has been updated, the system runs only small-scale parsl_csa.
If parsl_lca workflow has been updated, the system runs only small-scale lca_csa.
If deephyper_hpo workflow has been updated, the system runs only small-scale deephyper_hpo.
If improvelib has been updated, the system runs all workflows at small-scale.
